File: src/unity/Runtime/Ads/Internal/LockCapper.cs
namespace EE.Internal {
    internal class LockCapper : ICapper {
        private bool _locked;
        public bool IsCapped => _locked;

        public void Cap() {
            _locked = false;
        }

        public void Lock() {
            _locked = true;
        }

        public void Unlock() {
            _locked = false;
        }
    }
}

File: src/cpp/ee/ads/internal/LockCapper.cpp
//
//  LockCapper.cpp
//  Pods
//
//  Created by eps on 6/2/21.
//

#include "ee/ads/internal/LockCapper.hpp"

namespace ee {
namespace ads {
using Self = LockCapper;

Self::LockCapper()
    : locked_(false) {}

Self::~LockCapper() = default;

bool Self::isCapped() const {
    return locked_;
}

void Self::cap() {}

void Self::lock() {
    locked_ = true;
}

void Self::unlock() {
    locked_ = false;
}
} // namespace ads
} // namespace ee
